app/ui/src/main/java/com/fsck/k9/activity/MessageCompose.java

```diff
@@ -1325,6 +1328,9 @@ private void processDraftMessage(MessageViewInfo messageViewInfo) {
         // See buildIdentityHeader(TextBody) for a detailed description of the composition of this blob.
         Map<IdentityField, String> k9identity = new HashMap<>();
         String[] identityHeaders = message.getHeader(K9.IDENTITY_HEADER);
+        if (identityHeaders.length == 0) {
+            identityHeaders = messageViewInfo.rootPart.getHeader(K9.IDENTITY_HEADER);
```

Comment:
Wouldn't it be better to use the inner header and only fall back to the outer one?

Reply:
Yes, slightly :)